src/components/grid-infinite-scroll/grid-infinite-scroll.tsx

```diff
@@ -0,0 +1,253 @@
+   */
+  @Prop() itemCount = 0;
+
+  @Prop({ context: "queue" })
+  queue!: QueueApi;
+  @Prop({ context: "enableListener" })
+  enableListener!: EventListenerEnable;
+
+  /**
+   * The threshold distance from the bottom
+   * of the content to call the `infinite` output event when scrolled.
+   * The threshold value can be either a percent, or
+   * in pixels. For example, use the value of `10%` for the `infinite`
+   * output event to get called when the user has scrolled 10%
+   * from the bottom of the page. Use the value `100px` when the
+   * scroll is within 100 pixels from the bottom of the page.
+   */
+  @Prop() threshold = "15%";
+
+  /**
+   * If `true`, the infinite scroll will be hidden and scroll event listeners
+   * will be removed.
+   *
+   * Set this to true to disable the infinite scroll from actively
+   * trying to receive new data while scrolling. This is useful
+   * when it is known that there is no more data that can be added, and
+   * the infinite scroll is no longer needed.
+   */
+  @Prop() disabled = false;
+
+  /**
+   * The position of the infinite scroll element.
+   * The value can be either `top` or `bottom`.
+   */
+  @Prop() position: "top" | "bottom" = "bottom";
+
+  /**
+   * Emitted when the scroll reaches
+   * the threshold distance. From within your infinite handler,
+   * you must call the infinite scroll's `complete()` method when
+   * your async operation has completed.
+   */
+  @Event() gxInfinite!: EventEmitter<void>;
+
+  componentWillLoad() {
+    this.itemCountChanged();
+  }
+
+  @Watch("itemCount")
+  public itemCountChanged() {
+    setTimeout(() => {
+      const containerHeight = this.scrollEl.offsetHeight;
+      const contentHeight = this.scrollEl.querySelector("div").offsetHeight;
+
+      if (contentHeight < containerHeight) {
+        this.gxInfinite.emit();
+      }
+    }, 100);
+  }
+
+  @Watch("disabled")
+  protected disabledChanged(val: boolean) {
+    if (this.disabled) {
+      this.isLoading = false;
+      this.isBusy = false;
+    }
+    this.enableScrollEvents(!val);
+  }
+
+  @Watch("threshold")
+  protected thresholdChanged(val: string) {
+    if (val.lastIndexOf("%") > -1) {
+      this.thrPx = 0;
+      this.thrPc = parseFloat(val) / 100;
+    } else {
+      this.thrPx = parseFloat(val);
+      this.thrPc = 0;
+    }
+  }
+
+  async componentDidLoad() {
+    const contentEl = this.el.closest(this.infiniteScrollContainer);
+    if (contentEl) {
+      this.scrollEl = contentEl as HTMLElement;
+      this.thresholdChanged(this.threshold);
+      this.enableScrollEvents(!this.disabled);
+      if (this.position === "top") {
+        this.queue.write(() => {
+          if (this.scrollEl) {
+            this.scrollEl.scrollTop =
+              this.scrollEl.scrollHeight - this.scrollEl.clientHeight;
+          }
+        });
+      }
+    }
+  }
+
+  componentDidUnload() {
+    this.scrollEl = undefined;
```

Comment:
We don't need to clear the reference to the element. This will be handled automatically when the object is disposed.